framework/encode/custom_vulkan_api_call_encoders.cpp

```diff
@@ -542,6 +550,161 @@ VKAPI_ATTR VkResult VKAPI_CALL CreateGraphicsPipelines(VkDevice
+            }
+            else
+            {
+                GFXRECON_LOG_WARNING("[capture]vkCreateGraphicsPipelines:link:libraryCount=%d,pLibraries[0]=%p,"
+                                     "pLibraries[1]=%p,pLibraries[2]=%p",
+                                     libraryInfo->libraryCount,
+                                     libraryInfo->pLibraries[0],
+                                     libraryInfo->pLibraries[1],
+                                     libraryInfo->pLibraries[2]);
+            }
+            VkDescriptorSetVariableDescriptorCountAllocateInfo linkInfo = {};
+            linkInfo.sType = VK_STRUCTURE_TYPE_DESCRIPTOR_SET_VARIABLE_DESCRIPTOR_COUNT_ALLOCATE_INFO;
+
+            linkInfo.descriptorSetCount = libraryInfo->libraryCount;
+            std::vector<uint32_t> indexes;
+            size_t                vertexIdx, shaderIdx, fragmentIdx, fragmentShaderIdx;
+            for (size_t i = 0; i < vertexInputPipelines.size(); i++)
+            {
+                if (libraryInfo->pLibraries[0] == vertexInputPipelines[i])
+                {
+                    vertexIdx = i;
+                }
+            }
+            if (vertexIdx >= vertexInputPipelines.size())
+                vertexIdx = vertexInputPipelines.size() - 1;
+            indexes.push_back(vertexIdx);
+
+            for (size_t j = 0; j < shadersPipelines.size(); j++)
+            {
+                if (libraryInfo->pLibraries[1] == shadersPipelines[j])
+                {
+                    shaderIdx = j;
+                }
+            }
+            if (shaderIdx >= shadersPipelines.size())
+                shaderIdx = shadersPipelines.size() - 1;
+            indexes.push_back(shaderIdx);
+
+            if (libraryInfo->libraryCount == 4)
+            {
+                for (size_t k = 0; k < fragmentShaderPipelines.size(); k++)
+                {
+                    if (libraryInfo->pLibraries[2] == fragmentShaderPipelines[k])
+                    {
+                        fragmentShaderIdx = k;
+                    }
+                }
+                if (fragmentShaderIdx >= fragmentShaderPipelines.size())
+                    fragmentShaderIdx = fragmentShaderPipelines.size() - 1;
+                indexes.push_back(fragmentShaderIdx);
+
+                for (size_t l = 0; l < fragmentOutputPipelines.size(); l++)
+                {
+                    if (libraryInfo->pLibraries[3] == fragmentOutputPipelines[l])
+                    {
+                        fragmentIdx = l;
+                    }
+                }
+                if (fragmentIdx >= fragmentOutputPipelines.size())
+                    fragmentIdx = fragmentOutputPipelines.size() - 1;
+                indexes.push_back(fragmentIdx);
+            }
+            else
+            {
+                for (size_t k = 0; k < fragmentOutputPipelines.size(); k++)
+                {
+                    if (libraryInfo->pLibraries[2] == fragmentOutputPipelines[k])
```

Comment:
If libraryCount is < 3, this will be an out-of-bounds error.  The spec also doesn't guarantee ordering of the libraries; the bit in `flags` specifies what subset is being specified. My guess is that this hardcodes a particular app's layout of libraries.